lib/ACL/ACLStorageWrapper.php

```diff
@@ -291,16 +291,20 @@ public function getETag($path): string|false {
 		return parent::getETag($path);
 	}
 
-	public function getDirectDownload($path): array|bool {
+	public function getDirectDownload($path): array|false {
 		if (!$this->checkPermissions($path, Constants::PERMISSION_READ)) {
 			return false;
 		}
 
 		return parent::getDirectDownload($path);
 	}
 
-	public function getDirectoryContent($directory): \Traversable {
-		foreach ($this->getWrapperStorage()->getDirectoryContent($directory) as $data) {
+	public function getDirectoryContent($directory): \Traversable|false {
+		$content = $this->getWrapperStorage()->getDirectoryContent($directory);
+		if ($content === false) {
+			return false;
+		}
+		foreach ($content as $data) {
```

Comment:
Still puzzled by this return false thing :thinking:

Reply:
I'll respond in the server PR